internal/ai/openrouter/openrouter.go

```diff
@@ -0,0 +1,243 @@
+package openrouter
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/blacktop/ipsw/internal/ai/utils"
+)
+
+const (
+	openrouterChatCompletionsEndpoint = "https://openrouter.ai/api/v1/chat/completions"
+	openrouterModelsEndpoint          = "https://openrouter.ai/api/v1/models"
+)
+
+// Config holds the configuration for the OpenRouter LLM API client
+type Config struct {
+	Prompt      string  `json:"prompt"`
+	Model       string  `json:"model"`
+	Temperature float64 `json:"temperature"`
+	TopP        float64 `json:"top_p"`
+	Stream      bool    `json:"stream"`
+}
+
+// OpenRouter represents a client for the OpenRouter API
+type OpenRouter struct {
+	ctx    context.Context
+	conf   *Config
+	client *http.Client
+	models map[string]string
+}
+
+// chatMessage represents a single message in the conversation
+type chatMessage struct {
+	Role    string `json:"role"`
+	Content string `json:"content"`
+}
+
+// chatRequest is the payload sent to the OpenRouter chat API
+type chatRequest struct {
+	Model       string        `json:"model"`
+	Messages    []chatMessage `json:"messages"`
+	Temperature float64       `json:"temperature"`
+	TopP        float64       `json:"top_p"`
+	Stream      bool          `json:"stream"`
+}
+
+// chatResponse represents the response from the OpenRouter chat API
+type chatResponse struct {
+	ID      string `json:"id"`
+	Object  string `json:"object"`
+	Created int64  `json:"created"`
+	Model   string `json:"model"`
+	Choices []struct {
+		Message struct {
+			Role    string `json:"role"`
+			Content string `json:"content"`
+		} `json:"message"`
+		FinishReason string `json:"finish_reason"`
+	} `json:"choices"`
+	Usage struct {
+		PromptTokens     int `json:"prompt_tokens"`
+		CompletionTokens int `json:"completion_tokens"`
+		TotalTokens      int `json:"total_tokens"`
+	} `json:"usage"`
+}
+
+// modelsResponse represents the response from the OpenRouter models API
+type modelsResponse struct {
+	Data []struct {
+		ID              string  `json:"id"`
+		Name            string  `json:"name"`
+		Description     string  `json:"description"`
+		Context_length  int     `json:"context_length"`
+		PricePrompt     float64 `json:"pricing.prompt"`
+		PriceCompletion float64 `json:"pricing.completion"`
+	} `json:"data"`
+}
+
+// NewOpenRouter creates a new OpenRouter API client
+func NewOpenRouter(ctx context.Context, conf *Config) (*OpenRouter, error) {
+	client := &OpenRouter{
+		ctx:    ctx,
+		conf:   conf,
+		client: &http.Client{Timeout: 300 * time.Second},
+		models: make(map[string]string),
+	}
+
+	// Verify API key is available
+	_, err := client.getAPIKey()
```

Comment:
i think client.getAPIKey only needs to be called here and then can be added to the `OpenRouter` structure as a private member